File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums, target):
        lookup = {}  # value -> index
        
        for i in range(len(nums)):
            need = target - nums[i]
            
            if need in lookup:
                return [lookup[need], i]
            
            lookup[nums[i]] = i

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        vector<int> index;
        for(int i=0;i<n-1;i++)
        {
            for(int j=i+1;j<n;j++)
            {
                if(nums[i]+nums[j]==target)
                {
                    index.push_back(i);
                    index.push_back(j);
                }
            }
        }
        return index;
        
    }
};